### PR TITLE
Dither analysis

### PR DESCRIPTION
**Pull request by julienguy**
This PR adds some documentation and a script to perform a dither analysis with desimeter.
The most recent dither sequence from 2020/03/15 is used as an example.

On this particular sequence, the result is not great, the rms of residuals is of 0.5 arcsec (there is still work to be done, in particular validate the ADC rotation correction: signs and zero points).

Figure_1

### CODE REVIEW

**Comment by julienguy** [comment]
I had forgotten the option `--zbfit` to refit the Zhao-Burge polynomial coefficients from the fiducials.
With this, the desimeter-dither rms reduces to 0.42 arcsec which is a bit better. I think this indicates we have to work more on the `fvc2fp` transform, and in particular as a function of the ADC angles.

Figure_1

**Comment by schlafly** [comment]
Just verifying---this is without the "cheat" refitting rotation + scale + offset parameters?  The trends are clearly very low order.

**Comment by julienguy** [comment]
Yes, 'this is without the "cheat" refitting rotation + scale + offset parameters'.

File: bin/plot_dither_vs_desimeter [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+# Eddie's dithers
+#######################################################################
+t=fitsio.read(dither_file)
+print(t.dtype.names)
+if args.expid is None:
+    i=0
+else:
+    ind = np.flatnonzero(t['expid'][0, :] == args.expid)
+    if len(ind) > 0:
```
Comment (schlafly) [inline bin/plot_dither_vs_desimeter:68]: For what it's worth, I would treat this case as an error.

**Review by schlafly — APPROVED** [review]
This all looks good to me.  I'm slightly confused at the relationship with your
https://github.com/desihub/desicmx/blob/master/analysis/desimeter/plot_dither_vs_desimeter_matched.py
The chief difference seems to me to be that that analysis includes a 'cheat' to fix up an offset and scale and rotation parameters.  I understand that it's important not to use that cheat in practice, but the fact that with the cheat one obtains ~PM level performance seems really important to me, since the cheat parameters come entirely from the astrometric fit (I think?) and not from the FVC fit.

**Comment by julienguy** [comment]
- expid is now a required argument.
 - I added back the final 'cheat' transformation results because this is indeed useful information.
 - this replaces the desicmx script because I think it is simpler to have all desimeter related codes in the same place, I will remove the desicmx script to avoid confusion.

dither-55670

```
number of valid targets from dither analysis = 1758
residual scale along HA       = 1-7.8e-05
residual scale along Dec      = 1-4.1e-06
residual pointing offset dHA  = -0.11 arcsec
residual pointing offset dDec = -0.15 arcsec
residual rotation             = -9.19 arcsec
   desimeter(guide+fvc)-target rms 2D = 0.44 arcsec
                 dither-target rms 2D = 0.23 arcsec
   desimeter(guide+fvc)-dither rms 2D = 0.42 arcsec
transformed(desimeter) -dither rms 2D = 0.26 arcsec
```

**Pull request merged**